FNPlugin/Collectors/SolarWindCollector.cs

```diff
@@ -141,41 +156,63 @@ public override void OnFixedUpdate()
 
+            return star;
+        }
 
         // calculates solar wind concentration
-        private double CalculateSolarWindConcentration()
+        private static double CalculateSolarWindConcentration(double flux)
         {
-            // check if the vessel is not in shade
-            if (!PluginHelper.lineOfSightToSun(vessel))
-            {
-                strCollectingStatus = "No solar wind detected";
-                return dConcentrationSolarWind = 0.0;
-            }
+            double dAvgKerbinSolarFlux = 1409.285; // this seems to be the average flux at Kerbin just above the atmosphere (from my tests)
+            double dAvgSolarWindPerCubM = 6000.0; // various sources differ, most state that there are around 6 particles per cm^3, so around 6000 per m^3 (some sources go up to 10/cm^3 or even down to 2/cm^3, most are around 6/cm^3).
 
-            dConcentrationSolarWind = dKerbinDistance / CalculateDistanceToSun(); // In Kerbin vicinity, the solar wind concentration will be around 100% and it will rise the closer the vessel gets to the sun
-            return dConcentrationSolarWind;
+            double dConcentration = (flux / dAvgKerbinSolarFlux) * dAvgSolarWindPerCubM;
+            return dConcentration;
         }
 
         // calculates the distance to sun
-        private double CalculateDistanceToSun()
+        private static double CalculateDistanceToSun(Vector3d vesselPosition, Vector3d sunPosition)
         {
-            Vector3d sunPosition = FlightGlobals.fetch.bodies[0].position;
-            Vector3d vesselPosition = this.part.transform.position;
-            return dDistanceFromStar = Vector3d.Distance(vesselPosition,sunPosition);
+            double dDistance = Vector3d.Distance(vesselPosition,sunPosition);
+            return dDistance;
+        }
+
+        // helper function for readying the distance for the GUI
+        private string UpdateDistanceInGUI()
+        {
+            string distance = ((CalculateDistanceToSun(part.transform.position, localStar.transform.position) - localStar.Radius) / 1000).ToString("F2") + " km";
+            return distance;
         }
 
         // the main collecting function
         private void CollectSolarWind(double deltaTimeInSeconds, bool offlineCollecting)
         {
-            dConcentrationSolarWind = CalculateSolarWindConcentration();
+            dConcentrationSolarWind = CalculateSolarWindConcentration(part.vessel.solarFlux);
 
             string strSolarWindResourceName = InterstellarResourcesConfiguration.Instance.SolarWind;
             double dPowerRequirementsMW = (double)PluginHelper.PowerConsumptionMultiplier * mwRequirements; // change the mwRequirements number in part config to change the power consumption
 
-            // check for free space in solar wind 'tanks'
+            // checks for free space in solar wind 'tanks'
             dSolarWindSpareCapacity = part.GetResourceSpareCapacity(strSolarWindResourceName);
 
```

Comment:
Well, the resource definition I am seeing is (WarpPlugin/Resources/ResourceDefs.cfg):
RESOURCE_DEFINITION
{
	name = SolarWind
	density = 0.0002
	flowMode = ALL_VESSEL
	transfer = PUMP
	isTweakable = true
    	isVisible = true
	unitCost = 0.001
	volume = 1
}
And according to http://forum.kerbalspaceprogram.com/index.php?/topic/60254-liquid-fuel-density/, the density parameter is in tonnes per liter (hopefully metric tons and not US tons or whatever, as this would be equivalent to kg/cm3). Which would mean that the SolarWind's density is currently 0.0002 t/l = 0.0002 kg/cm3 = **0.2 g/cm3**. For comparison purposes, Neon currently has the density 0.0000009002, which is **0.0009002 g/cm3**. So good catch, I didn't think of this, but even from this quick check, our SolarWind currently seems to be too dense. 

Most sources I've seen claim that solar wind's density is about 6 atoms per cm3, so roughly about 6 * 1.66 * 10^−27 kg / cm3 = 9,96 * 10^-27 or 0,00000000000000000000000000996 kg/cm3, or 0,00000000000000000000000996 g/cm3. So the realistic density in KSP numbers should probably be those 9.96 * 10^-27 or **0.00000000000000000000000000996**. This should still be within float type limits.

I would ask you to please check my math as I had a terribly long day today and may have goofed up somewhere :)

(BTW what does moving the definition to CRP configs entail? Can we just move the definition there and be done with it, or does someone else have to okay it first? If were taking this much time to define the resource correctly, we might as well share it with others via the CRP :D)